code/game/objects/structures/table_frames.dm

```diff
@@ -21,36 +21,48 @@
 	var/framestack = /obj/item/stack/rods
 	var/framestackamount = 2
 
-/obj/structure/table_frame/attackby(obj/item/I, mob/user, params)
-	if(I.tool_behaviour == TOOL_WRENCH)
-		to_chat(user, "<span class='notice'>You start disassembling [src]...</span>")
-		I.play_tool_sound(src)
-		if(I.use_tool(src, user, 30))
-			playsound(src.loc, 'sound/items/deconstruct.ogg', 50, TRUE)
-			deconstruct(TRUE)
-		return
 
-	var/obj/item/stack/material = I
-	if (istype(material))
-		if(material?.tableVariant)
+/obj/structure/table_frame/wrench_act(mob/living/user, obj/item/I)
+	to_chat(user, "<span class='notice'>You start disassembling [src]...</span>")
+	I.play_tool_sound(src)
+	if(!I.use_tool(src, user, 3 SECONDS))
+		return TRUE
+	playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
+	deconstruct(TRUE)
+	return TRUE
+
+
+/obj/structure/table_frame/attackby(obj/item/I, mob/user, params)
+	if(isstack(I))
+		var/obj/item/stack/material = I
+		if(material.tableVariant)
 			if(material.get_amount() < 1)
 				to_chat(user, "<span class='warning'>You need one [material.name] sheet to do this!</span>")
 				return
+			if(locate(/obj/structure/table) in loc)
+				to_chat(user, "<span class='warning'>There's already a table built here!</span>")
+				return
 			to_chat(user, "<span class='notice'>You start adding [material] to [src]...</span>")
-			if(do_after(user, 20, target = src) && material.use(1))
-				make_new_table(material.tableVariant)
+			if(!do_after(user, 2 SECONDS, target = src) || !material.use(1) || (locate(/obj/structure/table) in loc))
+				return
+			make_new_table(material.tableVariant)
```

Comment:
This hurts, but it's what must be done. I fucking hate machine/structure stacking.